File: sdk/java/examples/Quickstart.java
import com.yowyob.loyalty.sdk.ApiException;
import com.yowyob.loyalty.sdk.LoyaltyClient;

import java.util.List;
import java.util.Map;

/**
 * Démarrage rapide — après `mvn install` du SDK, ajoutez la dépendance
 * com.yowyob:loyalty-sdk:1.0.0 à votre projet puis :
 */
public class Quickstart {

    public static void main(String[] args) {
        // 1. Initialisation — récupérez ces clés dans Portail > Applications
        LoyaltyClient loyalty = new LoyaltyClient(
                envOr("LOYALTY_PUBLIC_KEY", "pk_test_xxx"),
                envOr("LOYALTY_PRIVATE_KEY", "sk_test_xxx"),
                envOr("LOYALTY_BASE_URL", "http://localhost:8081"),
                System.getenv("LOYALTY_WEBHOOK_SECRET"));

        String memberId = "00000000-0000-0000-0000-000000000123";

        // 2. Envoyer un événement d'achat (avec clé d'idempotence)
        try {
            Map<String, Object> result = loyalty.trackEvent(
                    "purchase.completed", memberId, null, Map.of("amount", 4990), "commande-2026-0042");
            System.out.println("Événement traité : " + result.get("eventId"));
            @SuppressWarnings("unchecked")
            List<Map<String, Object>> effects = (List<Map<String, Object>>) result.get("effectsApplied");
            for (Map<String, Object> effect : effects) {
                System.out.println(" → effet : " + effect.get("effectType") + " (règle : " + effect.get("ruleName") + ")");
            }
        } catch (ApiException e) {
            System.err.println("Erreur API (" + e.getStatusCode() + ") : " + e.getMessage());
        }

        // 3. Consulter le solde de points du membre
        Map<String, Object> points = loyalty.getMemberPoints(memberId);
        System.out.println("Points disponibles : " + points.get("availablePoints"));

        // 4. Dans votre contrôleur de callback (ex. Spring) :
        //    Map<String, Object> event = loyalty.checkCallbackIntegrity(headers, rawBody);
        //    → lève SignatureVerificationException si la requête est forgée ; répondez alors 400.
    }

    private static String envOr(String name, String fallback) {
        String value = System.getenv(name);
        return value != null && !value.isBlank() ? value : fallback;
    }
}

File: sdk/php/examples/quickstart.php
<?php

require __DIR__ . '/../vendor/autoload.php';

use Yowyob\Loyalty\LoyaltyClient;
use Yowyob\Loyalty\Exception\ApiException;
use Yowyob\Loyalty\Exception\SignatureVerificationException;

// 1. Initialisation — récupérez ces clés dans Portail > Applications
$loyalty = new LoyaltyClient(
    getenv('LOYALTY_PUBLIC_KEY') ?: 'pk_test_xxx',
    getenv('LOYALTY_PRIVATE_KEY') ?: 'sk_test_xxx',
    getenv('LOYALTY_BASE_URL') ?: 'http://localhost:8081',
    getenv('LOYALTY_WEBHOOK_SECRET') ?: null
);

// 2. Envoyer un événement d'achat (avec clé d'idempotence)
try {
    $result = $loyalty->trackEvent(
        'purchase.completed',
        '00000000-0000-0000-0000-000000000123',
        null,
        ['amount' => 4990],
        'commande-2026-0042'
    );
    echo "Événement traité : " . $result['eventId'] . PHP_EOL;
    foreach ($result['effectsApplied'] as $effect) {
        echo " → effet : " . $effect['effectType'] . " (règle : " . $effect['ruleName'] . ")" . PHP_EOL;
    }
} catch (ApiException $e) {
    echo "Erreur API (" . $e->getStatusCode() . ") : " . $e->getMessage() . PHP_EOL;
}

// 3. Consulter le solde de points du membre
$points = $loyalty->getMemberPoints('00000000-0000-0000-0000-000000000123');
echo "Points disponibles : " . $points['availablePoints'] . PHP_EOL;

// 4. Dans votre contrôleur de callback (ex. /loyalty/callback) :
//    $payload = $loyalty->checkCallbackIntegrity(getallheaders(), file_get_contents('php://input'));
//    → lève SignatureVerificationException si la requête est forgée ; répondez alors 400.
